### DISCUSSION TITLE
[feature request] Excluding pages from build

### DISCUSSION THREAD

**Discussion by OlehDutchenko** [vercel/next.js#17271] [Ideas]
# Feature request

Excluding pages from build

## Is your feature request related to a problem? Please describe.

I want to have some tool to exlude some unwanted pages on build.

My use case:
I have demo page: `pages/ui-kit`. 
This page for developers and this route is availible on our testing platform.
But I do not need this route on production domain. And I do not need unnecessary code of this examples on production

## Describe the solution you'd like

It would be great to have some field in `next.config.js`  for excluding unwanted pages and ignore them (and their codebase) in build.

example:

```js
// next.config.js
module.exports = {
    excludePagesOnBuild: [
        'pages/ui-kit',
        'pages/other/unwanted-page'
    ]
}
```

## Describe alternatives you've considered

My solution for now is `redirect` field in `next.config.js`  
But still the final build will contain unnecessary files of examples from `ui-kit`.

**Comment by lfades** [comment]
You could run a script that removes unwanted pages before building the app

**Reply by OlehDutchenko** [reply]
Remove? No. I need this pages for develop project.
I already had an idea to cut this files into some temp folder or remove them at all. And rollback them after build. But it's hard process and if some step fall - it can break full stack of building and deployment. So I gave up on this solution

**Comment by Vadorequest** [comment]
I have a similar need with "template" pages, meant for developers to create a new page faster with proper defaults and documentation.

I do not need to build those pages, ever, and they shouldn't be available online.
